syncapi/storage/shared/syncserver.go

```diff
@@ -582,6 +582,7 @@ func (d *Database) addTypingDeltaToResponse(
 			jr.Ephemeral.Events = append(jr.Ephemeral.Events, ev)
 			res.Rooms.Join[roomID] = jr
 		}
+		res.NextBatch.TypingPosition = types.StreamPosition(d.EDUCache.GetLatestSyncPosition())
```

Comment:
Does this not have the chance to race (a call to `GetTypingUsersIfUpdatedAfter` vs a call to `GetLatestSyncPosition`)?